pkg/volume/csi/csi_mounter.go

```diff
@@ -273,7 +276,13 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 		return err
 	}
 
-	if err := csi.NodeUnpublishVolume(ctx, volID, dir); err != nil {
+	nodeUnpublishCredentials := map[string]string{}
+	if csiSource.NodePublishSecretRef != nil {
+		sn := csiSource.NodePublishSecretRef.Name
+		ns := c.pod.ObjectMeta.Namespace
```

Comment:
NodePublishSecretRef.Namespace

Reply:
Will be passing SecreteReference to getCredentialsFromSecret.